Sources/LanguageServerProtocol/Language.swift

```diff
@@ -11,23 +11,18 @@
 //===----------------------------------------------------------------------===//
 
 /// A source code language identifier, such as "swift", or "objective-c".
-public enum Language: String, Codable, Hashable {
-  case c
-  case cpp // C++, not C preprocessor
-  case objective_c = "objective-c"
-  case objective_cpp = "objective-cpp"
-  case swift
+public struct Language: RawRepresentable, Codable, Equatable, Hashable {
+  public typealias LanguageId = String
```

Comment:
Is it useful to have the extra type `Language.LanguageId`, or should we just use `RawValue` which is needed anyway?

Reply:
I like to cover generic types with more descriptive alias, but I can ditch it if you prefer not

Reply:
Okay, doesn't make much difference to me.